schwa/analysis/repository_analytics.py

```diff
@@ -307,4 +308,22 @@ def to_dict(self, name):
         metrics_dict = super().to_dict()
         metrics_dict["type"] = "method"
         metrics_dict["name"] = name
+        return metrics_dict
+
+
+class LineAnalytics(Metrics):
+    """ A class to represent Line Analytics
+
+    It the leaf of analytics.
+    """
+    def __init__(self):
+        super().__init__()
+
+    def compute_defect_probability(self):
+        self.defect_prob = self.defect_probability()
+
+    def to_dict(self, name):
+        metrics_dict = super().to_dict()
+        metrics_dict["type"] = "line"
+        metrics_dict["name"] = name
```

Comment:
instead of name we need to know the line number. Eventually also the path of the file to be more explicit

Reply:
Ideally, I would like to represent three different types of lines:
- Lines of code that can be in a file but not in a class or method (e.g., import statement).
- Lines of code that can be in a file and in a class but not in a method (e.g., a class' field).
- Lines of code that can be in a file and in a class and in a method.

I believe commit [0a76908](https://github.com/andrefreitas/schwa/pull/5/commits/0a7690853c549a9b3cd0b0689167ca769df1c00c) implements that, but I'm still not sure whether the `diff` method in the `java_parser.py` file is correctly implemented.  Could you please check that?